rawfile/filesystem/xfs.py

```diff
@@ -0,0 +1,93 @@
+from .base import (
+    FileSystem as FileSystemBase,
+    FileSystemFormatError,
+    FileSystemMountError,
+    FileSystemUnmountError,
+    FileSystemResizeError,
+    InvalidMountpointError,
+)
+from pathlib import Path
+import subprocess
+from .utils import get_device_for_mountpoint
+
+
+class XFS(FileSystemBase):
+    @property
+    def __filesystem__(self) -> str:
+        return "xfs"
+
+    def format_fs(self, options: list[str] = []) -> str:
+        try:
+            output = subprocess.run(
+                f"mkfs.{self.__filesystem__} {' '.join(options)} {self.device}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+            return output.stdout.decode()
+        except subprocess.CalledProcessError as e:
+            raise FileSystemFormatError(self.__filesystem__, e.stderr.decode())
+        except Exception as e:
+            raise FileSystemFormatError(self.__filesystem__, str(e))
+
+    def mount(
+        self,
+        mountpoint: str | None = None,
+        options: list[str] = [],
+    ) -> str:
+        if mountpoint is None:
```

Comment:
there's a lot of code that seems duplicated, I wonder if we could have a default implementation that we could reuse as xfs/ext4 code would be mostly the same

Reply:
Yeah, will fix that too!

Reply:
Shared logics are moved into the base class